databroker/experimental/server_ext.py

```diff
@@ -0,0 +1,501 @@
+        principal=UNCHANGED,
+        **kwargs,
+    ):
+        if metadata is UNCHANGED:
+            metadata = self.metadata
+        if queries is UNCHANGED:
+            queries = self.queries
+        if sorting is UNCHANGED:
+            sorting = self.sorting
+        if principal is UNCHANGED:
+            principal = self.principal
+        return type(self)(
+            database=self.database,
+            directory=self.directory,
+            metadata=metadata,
+            queries=queries,
+            sorting=sorting,
+            access_policy=self.access_policy,
+            principal=principal,
+            **kwargs,
+        )
+
+    def post_metadata(self, metadata, structure_family, structure, specs):
+
+        mime_structure_association = {
+            StructureFamily.array: "application/octet-stream",
+            StructureFamily.dataframe: APACHE_ARROW_FILE_MIME_TYPE,
+        }
+
+        uid = str(uuid.uuid4())
+
+        validated_document = Document(
+            uid=uid,
+            structure_family=structure_family,
+            structure=structure,
+            metadata=metadata,
+            specs=specs,
+            mimetype=mime_structure_association[structure_family],
+        )
+
+        # After validating the document must be encoded to bytes again to make it compatible with MongoDB
+        if validated_document.structure_family == StructureFamily.dataframe:
+            validated_document.structure.micro.meta = bytes(
+                serialize_arrow(validated_document.structure.micro.meta, {})
+            )
+
+        self.collection.insert_one(validated_document.dict())
+        return uid
+
+    def authenticated_as(self, identity):
+        if self.principal is not None:
+            raise RuntimeError(f"Already authenticated as {self.principal}")
+        if self.access_policy is not None:
+            raise NotImplementedError("No support for Access Policy")
+        return self
+
+    def _build_mongo_query(self, *queries):
+        combined = self.queries + list(queries)
+        if combined:
+            return {"$and": combined}
+        else:
+            return {}
+
+    def __getitem__(self, key):
+        query = {"uid": key}
+        doc = self.collection.find_one(self._build_mongo_query(query), {"_id": False})
+        if doc is None:
+            raise KeyError(key)
+
+        if doc["structure_family"] == StructureFamily.array:
+            return WritingArrayAdapter(self.collection, self.directory, doc)
+        elif doc["structure_family"] == StructureFamily.dataframe:
+            return WritingDataFrameAdapter(self.collection, self.directory, doc)
+        else:
+            raise ValueError("Unsupported Structure Family value in the databse")
+
+    def __iter__(self):
+        # TODO Apply pagination, as we do in Databroker.
+        for doc in list(
+            self.collection.find(
+                # self._build_mongo_query({"active": True}), {"uid": True}
```

Comment:
why is this commented out? The below solution relies on `data_url` being defined which is not as generic as the document model which allows either `data_url` or `data_blob`.

Reply:
should `data_blob` be removed from `Document`?

Reply:
That was dragged from the initial draft. Now, that I made enough test with the new document for the pydantic model, we are not using the active parameter anymore. I will remove this comment and other like these because they are outdated and not useful for this new approach